watcher/lib/resources/resources.go

```diff
@@ -0,0 +1,377 @@
+	Email string
+}
+
+// String returns the target's string representation.
+func (t AlertTarget) String() string {
+	return fmt.Sprintf("AlertTarget(Email=%v)", t.Email)
+}
+
+// Alert represents a monitoring alert.
+type Alert struct {
+	// CRDName is the name of PrometheusRule custom resource.
+	CRDName string
+	// AlertName is the alerting rule name.
+	AlertName string
+	// GroupName is group name the alert belongs to.
+	GroupName string
+	// Formula is the alert expression.
+	Formula string
+	// Delay is an optional delay before alert should be triggerred.
+	Delay time.Duration
+	// Labels is the labels that get attached to the alert.
+	Labels map[string]string
+	// Annotations are used to attach longer information to the alert.
+	Annotations map[string]string
+}
+
+// String returns the alert's string representation.
+func (a Alert) String() string {
+	return fmt.Sprintf("Alert(CRDName=%v,AlertName=%v,GroupName=%v,Formula=%v,Delay=%v,Labels=%v)",
+		a.CRDName, a.AlertName, a.GroupName, a.Formula, a.Delay, a.Labels)
+}
+
+// UpdateSMTPConfig updates cluster SMTP configuration.
+func (c *Client) UpdateSMTPConfig(smtpConf SMTPConfig) error {
+	c.Infof("Updating SMTP configuration: %s.", smtpConf)
+	conf, err := c.getAlertmanagerConfig()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	err = updateSMTPConfig(conf, fmt.Sprintf("%v:%v", smtpConf.Host, smtpConf.Port),
+		smtpConf.Username, smtpConf.Password)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	err = c.updateAlertmanagerConfig(conf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// DeleteSMTPConfig resets cluster SMTP configuration.
+func (c *Client) DeleteSMTPConfig() error {
+	c.Info("Deleting SMTP configuration.")
+	conf, err := c.getAlertmanagerConfig()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	err = updateSMTPConfig(conf, "", "", "")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	err = c.updateAlertmanagerConfig(conf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// UpdateAlertTarget updates recipient of monitoring alerts.
+func (c *Client) UpdateAlertTarget(alertTarget AlertTarget) error {
+	c.Infof("Updating alert target: %s.", alertTarget)
+	conf, err := c.getAlertmanagerConfig()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// We expect "default" receiver to be present.
+	defaultReceiver, err := getDefaultReceiver(conf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Update its email config.
+	defaultReceiver.EmailConfigs = []*EmailConfig{
+		{To: alertTarget.Email},
+	}
+	err = c.updateAlertmanagerConfig(conf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// DeleteAlertTarget resets monitoring alerts recipient.
+func (c *Client) DeleteAlertTarget() error {
+	c.Info("Deleting alert target.")
+	conf, err := c.getAlertmanagerConfig()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// We expect "default" receiver to be present.
+	defaultReceiver, err := getDefaultReceiver(conf)
```

Comment:
It feels that with this requirement, lib/resources#Config should contain the default receiver as a dedicated field and 
getAlertmanagerConfig should populate this field or give an error if not found. 
Then it would not look scary when updating the reference field:
```go
conf.DefaultReceiver.EmailConfigs = nil
```

Reply:
The `Config` struct maps directly onto the Alertmanager configuration file so it can't have a top-level "DefaultReceiver" field (w/o hacking the struct anyway, which I'd like to avoid). This default receiver is always present in the monitoring app configuration.